### PR TITLE
TF: Update multiple choice example

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

Updates the TF example for multiple choice.

This example update is not as clean as the previous ones -- despite the deletion of a custom dataset function, it required the addition of a custom data collator ([just like in its PT counterpart](https://github.com/huggingface/transformers/blob/master/examples/pytorch/multiple-choice/run_swag.py#L150)).

### CODE REVIEW

File: examples/tensorflow/multiple-choice/run_swag.py [hunk 1]
```diff
@@ -65,51 +67,61 @@ def on_epoch_end(self, epoch, logs=None):
         self.model.save_pretrained(self.output_dir)
 
 
-def convert_dataset_for_tensorflow(
-    dataset, non_label_column_names, batch_size, dataset_mode="variable_batch", shuffle=True, drop_remainder=True
-):
-    """Converts a Hugging Face dataset to a Tensorflow Dataset. The dataset_mode controls whether we pad all batches
-    to the maximum sequence length, or whether we only pad to the maximum length within that batch. The former
-    is most useful when training on TPU, as a new graph compilation is required for each sequence length.
+@dataclass
```
Comment (gante) [inline examples/tensorflow/multiple-choice/run_swag.py:70]: copy/paste from PT, except on L117, 121, and 123.

File: examples/tensorflow/multiple-choice/run_swag.py [hunk 2]
```diff
@@ -65,51 +67,61 @@ def on_epoch_end(self, epoch, logs=None):
...
+
+            * :obj:`True` or :obj:`'longest'`: Pad to the longest sequence in the batch (or no padding if only a single
+              sequence if provided).
+            * :obj:`'max_length'`: Pad to a maximum length specified with the argument :obj:`max_length` or to the
+              maximum acceptable input length for the model if that argument is not provided.
+            * :obj:`False` or :obj:`'do_not_pad'` (default): No padding (i.e., can output a batch with sequences of
+              different lengths).
+        max_length (:obj:`int`, `optional`):
+            Maximum length of the returned list and optionally padding length (see above).
+        pad_to_multiple_of (:obj:`int`, `optional`):
```
Comment (sgugger) [inline examples/tensorflow/multiple-choice/run_swag.py:90]: Note that we write the doc differently now, you should use Markdown instead (I know the PyTorch example needs an update too, if you want to treat it in the same PR that would be awesome!)

**Review by sgugger — APPROVED** [review]
Thanks a lot for working on this!

File: examples/tensorflow/multiple-choice/run_swag.py (hunk 2 above)
Comment (gante) [reply examples/tensorflow/multiple-choice/run_swag.py:90]: Will update in this PR 👍

**Review by Rocketknight1 — APPROVED** [review]
With apologies for lateness, this is a good, straightforward PR to get rid of that legacy `convert_dataset_for_tensorflow` code. Thanks for working on it!

**Pull request merged**